### PR TITLE
Allow sql filter IN operator right-hand side values including whitespaces

### PR DESCRIPTION
**Pull request by WeichenXu123**
Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Allow sql filter IN operator right-hand side values including whitespaces, i.e.,
allow filter like `run_id IN ('v1', 'v2', 'v3')`,
before this PR, we have to write `run_id IN ('v1','v2','v3')` otherwise raises error.

## How is this patch tested?

Unit tests.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/search_utils.py [hunk 1]
```diff
@@ -653,14 +654,17 @@ def _check_valid_identifier_list(cls, value_token):
...
+                    cls.WHITESPACE_VALUE_TYPE
+                },
                 value_token._groupable_tokens[0].tokens,
             )
         ):
             raise MlflowException(
-                "While parsing a list in the query, expected string value "
-                "or punctuation, but got different type in list: {value_token}".format(
+                "While parsing a list in the query, expected string value or punctuation "
+                "or whitespace, but got different type in list: {value_token}".format(
```
Comment (harupy) [inline mlflow/utils/search_utils.py:667]:
```suggestion
                "While parsing a list in the query, expected string value, punctuation "
                "or whitespace, but got different type in list: {value_token}".format(
```

File: tests/store/model_registry/test_sqlalchemy_store.py [hunk 2]
```diff
@@ -876,12 +870,18 @@ def search_versions(filter_string):
...
+            set([1, 2, 3]),
+        )
+
         # search using the IN operator with bad lists should return exceptions
         with self.assertRaisesRegex(
             MlflowException,
             (
                 r"While parsing a list in the query, "
-                r"expected string value or punctuation, "
+                r"expected string value or punctuation or whitespace, "
```
Comment (harupy) [inline tests/store/model_registry/test_sqlalchemy_store.py:884]:
```suggestion
                r"expected string value, punctuation or whitespace, "
```

File: tests/store/model_registry/test_sqlalchemy_store.py [hunk 3]
```diff
@@ -876,12 +870,18 @@ def search_versions(filter_string):
             set([2, 3]),
         )
 
+        # search IN operator with whitespaces between rhs values
```
Comment (harupy) [inline tests/store/model_registry/test_sqlalchemy_store.py:873]:
```suggestion
        # search IN operator with right-hand side value containing spaces
```

Comment (harupy) [inline tests/store/model_registry/test_sqlalchemy_store.py:873]:
```suggestion
        # search IN operator with right-hand side value containing whitespaces
```

**Review by harupy — APPROVED** [review]
LGTM once the comments are addressed.

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @WeichenXu123 !

**Pull request merged**